torchdata/dataloader2/dataloader2.py

```diff
@@ -280,17 +292,20 @@ def state_dict(self) -> Dict[str, Any]:
 
```

Comment:
I think at it's current form, we want to encourage people to call `state_dict` at the beginning of the epoch? Or, we need to make sure nothing within each DataPipe's buffer is serialized (which I think is already true for now)? Thoughts?

The changes may come in subsequent PR